app/models/chat_channel_membership.rb

```diff
@@ -89,7 +93,11 @@ def other_user
   end
 
   def permission
-    errors.add(:user_id, "is not allowed in chat") if chat_channel.direct? && chat_channel.slug.split("/").exclude?(user.username)
+    return unless chat_channel
```

Comment:
Wouldn't it be invalid without a chat_channel? Is it necessary to even check this?

Reply:
I added it because in one of the tests it ended up triggering an error as chat_channel wasn't nil. I think it's because ordering could trigger one before the other?